### PR TITLE
6502: illegal/undocumented instructions

### PR DESCRIPTION
**Pull request by q6r**
Support to show illegal/undocumented 6502 instructions. R2 shows 'illegal' instead of the instruction, now it doesn't. IDA shows almost all the illegal/undocumented instructions except for 'hlt'.

info from : http://unusedino.de/ec64/technical/aay/c64/ibmain.htm
binary that contains the undocumented opcodes : http://0x80.org/illegal.bin

### CODE REVIEW

File: libr/asm/arch/6502/6502dis.c [hunk 1]
```diff
@@ -1,125 +1,163 @@
+/* radare - LGPL - Copyright 2015 - condret, Ricardo Quesada, qnix */
 #include <r_asm.h>
 #include <r_lib.h>
 #include <string.h>
 #include "../snes/snesdis.c"
 
+static struct {
```
Comment (jvoisin) [inline libr/asm/arch/6502/6502dis.c]: Isn't this struct `const`?

**Comment by condret** [comment]
let's wait what @ricardoquesada says about this. he has more knowledge about 6502 than i have about gameboy-z80

**Comment by ricardoquesada** [comment]
we should tell somehow which opcodes are the undocumented ones. One possibility is to add that info here: https://github.com/radare/radare2/blob/master/libr/asm/d/6502 like this:
```
lax=loads both the accumulator and the X register with the contents 
of a memory location (undocumented)
ane=bla bla bla  (undocumented, unstable)
```
I think we should do that, but that won't be enough since there are some undocumented opcodes that behave like `nop`... and we can't say that `nop` is undocumented, so it would be great to say somehow that `0x1a`, `0x3a`, etc are undocumented `nop`s, and say that `0xea` is the documented one.

In that regard, whatever we do, it should be consistent with the rest of radare.
@condret what is the best way to document illegal/undocumented opcodes? how are the illegal opcodes being treated in the rest of the platforms? Does radare have some kind of "auto-comments"? If so, we could add "; undocumented opcode" / "; undocumented,unstable opcode" as auto-comments.

Thoughts?

File: libr/asm/arch/6502/6502dis.c [hunk 2]
```diff
@@ -1,125 +1,163 @@
+/* radare - LGPL - Copyright 2015 - condret, Ricardo Quesada, qnix */
```
Comment (ricardoquesada) [inline libr/asm/arch/6502/6502dis.c]: I don't think I have contributed to that file... feel free to remove me. thanks.

**Comment by condret** [comment]
i think that adding it to t is great, since pd should only show disassembly, in a way that represents what the cpu does

**Comment by condret** [comment]
merged, thanks

**Pull request closed without merging**